### DISCUSSION TITLE
How to change Source for web static apps

### DISCUSSION THREAD

**Discussion by fndmeghanshannon** [Azure/static-web-apps#1093] [Q&A] [answered]
How do you change Source for Static Web Apps? When I set up my Static Web App, I gave it a GitHub Source. I then decided I needed to fully switch GitHub repositories. How is that done?

I know you can do so on regular web apps under deployment but deployment doesn't show under options for Static Web Apps. 

Thanks!

**Comment by thomasgauvin** [comment, marked as answer]
Thanks for the feedback @fndmeghanshannon , this is not currently available via the Azure Portal. You may do so via the Az CLI using the disconnect and reconnect commands: https://learn.microsoft.com/en-us/cli/azure/staticwebapp?view=azure-cli-latest#az-staticwebapp-reconnect https://learn.microsoft.com/en-us/cli/azure/staticwebapp?view=azure-cli-latest#az-staticwebapp-disconnect. Alternatively, you can create a new Static Web App in the Portal

**Reply by Runeloon** [reply]
#536 Currently the cli cannot update the source

**Reply by atemerev** [reply]
CLI cannot update the source (the bug is open since 2021), there is no way to do it from UI as well. Recreating the web app from scratch will mean redoing custom domain validation etc, which will take a lot of time. I am very much not happy about it.

**Comment by harshitsahu2311** [comment]
It is also not working

az staticwebapp reconnect -n Testing-web-app --source https://github.com/harshitsahu2311/Nodejs-Project.git -b main --login-with-github  

ERROR: Static Web App Testing-web-app already exists in resource group Demo-RG

**Reply by movalex** [reply]
same issue here. 
after disconnecting, cannot reconnect. 
```
az staticwebapp disconnect --name path-converter
az staticwebapp reconnect --name path-converter --source https://github.com/repo/repourl.git -b main --login-with-github
```
shows 
`Static Web App path-converter already exists in resource group...`

**Reply by stefanosambruna** [reply]
use `update` instead. 
`az staticwebapp update --name path-converter --source https://github.com/repo/repourl.git -b main --login-with-github`

**Reply by nshobert** [reply]
Thank you @stefanosambruna! I had the same problem with `reconnect`, but using `update` worked.

**Reply by martinguenther** [reply]
`--login-with-github` is not a supported param of the `update` command. Just use it without.

`az staticwebapp update --name path-converter --source https://github.com/repo/repourl.git -b main`

**Reply by jack-anglicarevic** [reply]
Using `update` and not specifying `--login-with-github` worked for me.

Once this runs, you should get some JSON returned in your console, and the UI re-appears in the Azure portal.

Note that you may need to specify subscription as well with the `--subscription` flag.
